cmd/kata-pkgsync/cli/main.go

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) 2019 SUSE LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/marcov/obsgo"
+	"github.com/mlafeldt/pkgcloud"
+	"github.com/sirupsen/logrus"
+	pb "gopkg.in/cheggaaa/pb.v1"
+)
+
+//nolint[:gochecknoglobals]
+var (
+	// empty variables are set by "go build -ldflags" option
+	name           = ""
+	version        = ""
+	commit         = ""
+	defaultConfig  = "config.yaml"
+	defaultOBSDest = "obs-packages"
+)
+
+func usage() {
+	fmt.Fprintf(flag.CommandLine.Output(), `NAME:
+   %s - Synchronize packages from OBS to Packagecloud
+
+USAGE:
+   %s [options] [config file]
+
+By default %s reads the configuration from a file named %s. This can be
+overridden by passing the path to a config file.
+
+Options:
+`, name, name, name, defaultConfig)
+
+	flag.PrintDefaults()
+}
+
+func getOBSProjects(cfgProjects map[string]CfgOBSProject) []obsgo.Project {
+	var projects []obsgo.Project
+
+	for n, p := range cfgProjects {
+		proj := obsgo.Project{
+			User:     p.Auth.User,
+			Password: p.Auth.Password,
+		}
+
+		if len(p.Archs) == 0 {
+			p.Archs = append(p.Archs, "")
+		}
+
+		if len(p.Releases) == 0 {
+			p.Releases = append(p.Releases, "")
+		}
+
+		// Kata projects names format is "project:release:architecture"
+		for _, arch := range p.Archs {
+			for _, release := range p.Releases {
+				var fullname strings.Builder
+
+				fullname.WriteString(n)
+				if arch != "" {
+					fmt.Fprintf(&fullname, ":%s", arch)
+				}
+				if release != "" {
+					fmt.Fprintf(&fullname, ":%s", release)
+				}
+
+				proj.Name = fullname.String()
+				projects = append(projects, proj)
+			}
+		}
+	}
+
+	return projects
+}
+
+func getXferBinaries(pkg obsgo.PackageInfo, pcDistro string,
+	pcPackages []pkgcloud.Package, pcPackagesNeeded *[]bool) []obsgo.PkgBinary {
+
+	var xferBins []obsgo.PkgBinary
+	for _, src := range pkg.Files {
+		found := false
+		for i, dst := range pcPackages {
+			logrus.Debugf("Checking pkg %s %s VS %s %s", src.Filename, pcDistro, dst.Filename, dst.DistroVersion)
```

Comment:
Nit: Not required but you might want to consider converting the logrus calls to make them structured for parity with other tools and Kata components. This call would become something like:

```go
logrus.WithFields(logrus.Fields{
    "source-file" : src.Filename,
    "source-distro" : pcDistro,
    "destination-file" : dst.Filename,
    "destination-distro" : dst.DistroVersion,
}).Debug("Checking package")
```